src/helper/remote_config/protocol/targets.hpp

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "path.hpp"
+
+namespace dds::remote_config::protocol {
+
+class targets {
```

Comment:
Honestly this class (and pretty much every other class inside `protocol/`) has no logic at all to it. Having getters isn't useful and provides no benefits if they are just providing access to the contents of the structure. The constructors are also generally quite simple. If I were you I'd just go back to having these as simple POD structs and for the comparison operators you can define them as follows:

`inline bool operator==(const targets& rhs, const targets& lhs) `

Or we can move to C++20 and the compiler will auto-generate them I think.

Reply:
Would you make them structs with constructor or not even that?

Reply:
I wouldn't add a constructor unless there's significant logic (which I haven't seen in any of them), the idea is that you create these objects with data in them in order to pass them to a serializer or as a result of a parser (deserializer), so they are just containers of data that simplify your interfaces but have no logic and therefore do not benefit from a constructor or RAII or anything of the sort.

Reply:
Ok

Reply:
Done